interfaces/prompting/requestrules/requestrules.go

```diff
@@ -461,6 +461,9 @@ func (rdb *RuleDB) addRulePermissionToTree(rule *Rule, permission string) []prom
 		switch {
 		case !exists:
 			newVariantEntries[variantStr] = newEntry
+		case conflictingVariantEntry.RuleID == rule.ID:
+			// Rule has duplicate variant, so ignore it
+			return
```

Comment:
Not sure why the new test `TestAddRuleDuplicateVariants` isn't covering this?? Investigating...